org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/contentassist/XbaseContentProposalPriorities.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2013, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xbase.ui.contentassist;
+
+import org.eclipse.jface.text.contentassist.ICompletionProposal;
+import org.eclipse.xtext.common.types.JvmExecutable;
+import org.eclipse.xtext.common.types.JvmField;
+import org.eclipse.xtext.common.types.JvmIdentifiableElement;
+import org.eclipse.xtext.ui.editor.contentassist.ConfigurableCompletionProposal;
+import org.eclipse.xtext.ui.editor.contentassist.ContentProposalPriorities;
+import org.eclipse.xtext.xbase.scoping.batch.IIdentifiableElementDescription;
+import org.eclipse.xtext.xbase.scoping.batch.SimpleIdentifiableElementDescription;
+import org.eclipse.xtext.xbase.scoping.batch.StaticFeatureDescriptionWithTypeLiteralReceiver;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ */
+public class XbaseContentProposalPriorities extends ContentProposalPriorities {
+
+	@Override
+	public void adjustCrossReferencePriority(ICompletionProposal proposal, String prefix) {
+		if (proposal instanceof ConfigurableCompletionProposal) {
+			ConfigurableCompletionProposal configurableProposal = (ConfigurableCompletionProposal) proposal;
+			Object desc = configurableProposal.getAdditionalData(XbaseProposalProvider.DESCRIPTION_KEY);
+			if (desc instanceof SimpleIdentifiableElementDescription) {
+				if (!"this".equals(configurableProposal.getReplacementString())
+						&& !"super".equals(configurableProposal.getReplacementString())) {
+					adjustPriority(proposal, prefix, 570);
+					return;
+				}
+			} else if (desc instanceof StaticFeatureDescriptionWithTypeLiteralReceiver) {
+				adjustPriority(proposal, prefix, 560);
```

Comment:
This will fall through to the super.adjust.. call. Not sure if that was the intention in the xtend code.

Reply:
PS - https://bit.ly/3bJ43Kg

Reply:
I think this was a bug in the old impl.